include/LeMonADE/feature/FeatureConnectionSc.h

```diff
@@ -0,0 +1,621 @@
+  *
+  * @param stream output-stream
+  * @param label object of class MonomerReactivity
+  * @return output-stream
+  **/
+  // stream overload leads to wrong file output
+ /* std::ostream& operator<< (std::ostream& stream, const MonomerReactivity & Reactivity)
+  {
+	  stream
+	  << Reactivity.isReactive() << "/"
+	  << Reactivity.getNMaxBonds();
+	  return stream;
+  };
+*/
+
+  /**
+  * @brief \b Stream \b In \b operator of the MonomerReactivity
+  *
+  * @details Streams in the elements. And setting at first chainID, then NLabels, at last LabelID.
+  *
+  * @param stream input-stream
+  * @param label object of class MonomerReactivity
+  * @return input-stream
+  **/
+// stream overload leads to wrong file input
+  std::istream& operator >> (std::istream& stream, MonomerReactivity & Reactivity)
+  {
+	  int temp;
+	  stream >> temp; Reactivity.setReactive(temp); stream.ignore(1);
+	  stream >> temp; Reactivity.setNMaxBonds(temp); 
+	  return stream;
+  }
+
+
+
+/*****************************************************************/
+/**
+ * @class ReadReactivity
+ *
+ * @brief Handles BFM-File-Reads \b !reactivity
+ * @tparam IngredientsType Ingredients class storing all system information.
+ */
+template < class IngredientsType>
+class ReadReactivity: public ReadToDestination<IngredientsType>
+{
+public:
+  ReadReactivity(IngredientsType& i):ReadToDestination<IngredientsType>(i){}
+  virtual ~ReadReactivity(){}
+  virtual void execute();
+};
+
+
+/*****************************************************************/
+/**
+ * @class WriteReactivity
+ *
+ * @brief Handles BFM-File-Write \b !reactivity
+ * @tparam IngredientsType Ingredients class storing all system information.
+ **/
+template <class IngredientsType>
+class WriteReactivity:public AbstractWrite<IngredientsType>
+{
+public:
+	//! Only writes \b !reactivity into the header of the bfm-file.
+  WriteReactivity(const IngredientsType& i)
+    :AbstractWrite<IngredientsType>(i){this->setHeaderOnly(true);}
+  virtual ~WriteReactivity(){}
+  virtual void writeStream(std::ostream& strm);
+};
+/*****************************************************************************/
+/**
+ * @file
+ * @date   2019/02/05
+ * @author Toni
+ *
+ * @class FeatureConnectionSc
+ * @brief This Feature add new bonds between monomers.
+ *
+ * @details Works only in combination with an excluded volume feature
+ *
+ * @tparam 
+ * */
+
+///////////////////////////////////////////////////////////////////////////////
+//DEFINITION OF THE CLASS TEMPLATE   	                                ///////
+//Implementation of the members below					///////
+///////////////////////////////////////////////////////////////////////////////
+
+class FeatureConnectionSc : public Feature {
+  
+public:
+	//! This Feature requires a monomer_extensions.
+	typedef LOKI_TYPELIST_1(MonomerReactivity) monomer_extensions;
+	
+	//! 
+// 	typedef LOKI_TYPELIST_1(FeatureExcludedVolumeSc<>) required_features_back;
```

Comment:
FeatureExcludedVolumeSc<> should be mandatory as this algorithm implicitly uses excluded volume condition

Reply:
this feature should also require the "Feature*UnsaveCheck" e.g. FeatureMoleculesIOUnsaveCheck.h, FeatureBondsetUnsaveCheck.
Is there any possibility to "overwrite" the "save" features in the LOKi-List. If not, maybe it's a good practice to implement a test in ConfigureSystem to check for incompatible feature and throw an error.